### PR TITLE
updating docs for readability (first review)

### PR DESCRIPTION
**Pull request by nilmeier**
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by aloknsingh** [comment]
Hi @nilmeier 

Thanks for this. But I am having merge pbs

`error: Failed to merge in the changes.
Patch failed at 0001 updating docs for readability (not complete)
The copy of the patch that failed is found in: .git/rebase-apply/patch
`

if you are here tommorrow we can sort it out together

File: R4ML/R/ml.model.svm.R [hunk 1]
```diff
@@ -150,24 +151,24 @@ setMethod("initialize", "r4ml.svm",
 #' @examples \dontrun{
 #' 
 #' # Load the Iris dataset 
-#' iris_hf <- as.r4ml.frame(iris)
+#' iris_r4mlDF <- as.r4ml.frame(iris)
```
Comment (aloknsingh) [inline R4ML/R/ml.model.svm.R]: I was thinking , r4mlDF kind of mix cases and also too long. I think  we can fix the doc bug and merge it in the IOP and then change _hf or _hm kind of code later.
what do you think?

or would it be easier to subsititute similar case and smaller name for r4mlDF with r4mldf or r4ml_df?

Comment (bdwyer2) [reply R4ML/R/ml.model.svm.R]: I don't think we should use `hf` or `hm`

File: R4ML/R/airline.R [hunk 2]
```diff
@@ -15,7 +15,11 @@
 #
 #' Airline Dataset
 #'
-#' A 1\% sample of the "airline" dataset available at
+#' A 1\% sample of the "airline"
+#' 
+#' 
+#'
```
Comment (bdwyer2) [inline R4ML/R/airline.R]: Why did you add all these spaces?

File: R4ML/R/ml.model.pca.R [hunk 3]
```diff
@@ -30,24 +30,24 @@ setClass(
...
 #' Default is set to FALSE.
-#' @param scale (logical) a boolean value indicating whether all columns in the data should be scaled to have variance one.
+#' @param scale (logical):  A boolean value indicating whether all columns in the data should be scaled to have variance one.
 #' Default is set to FALSE.
-#' @param projData (logical) a boolean value indicating whether the data should be projected in the new feature space. If
+#' @param projData (logical):  A boolean value indicating whether the data should be projected in the new feature space. If
 #' set to TRUE, a list (r4ml.pca model, r4ml.matrix projData) is returned. If set to FALSE, only the r4ml.pca model is returned.
 #' Default value for \code{projData} is TRUE.
-#' @param applyPCA (r4ml.pca) (optional) Currently is not supported. In future this option will import an existing r4ml.pca model. If provided, \code{data} will be projected to the
+#' @param (optional) applyPCA (r4ml.pca):  Currently is not supported. In future this option will import an existing r4ml.pca model. If provided, \code{data} will be projected to the
```
Comment (bdwyer2) [inline R4ML/R/ml.model.pca.R]: I think `(optional)` should come after `applyPCA`

File: R4ML/R/r4ml.frame.R [hunk 4]
```diff
@@ -97,21 +97,20 @@ r4ml.calc.num.partitions <- function(object_size) {
 }
 
 
-#' Coerce to a R4ML Frame
+#' Coerce to an R4ML Frame
 #'
-#' Convert a data.frame, r4ml.matrix, or Spark DataFrame into a r4ml.frame
+#' Convert a data.frame, r4ml.matrix, or Spark DataFrame into an r4ml.frame
```
Comment (bdwyer2) [inline R4ML/R/r4ml.frame.R:102]: In other places I think we use `SparkDataFrame` instead of `Spark DataFrame`. I think we should try to be consistent about this.

File: R4ML/R/sysml.bridge.R [hunk 5]
```diff
@@ -32,8 +32,8 @@ requireNamespace("SparkR")
 #' @field ncol number of col of the block matrix
 #' @field bnrow number of row in one block
 #' @field bncol number of col of one block
-#' @field env environment which among other things contains the underlying
-#'  java reference
+#' @field env An R environment that stores bookkeeping states of the class,
+#'        along with java refience corresponding to the JVM.
```
Comment (bdwyer2) [inline R4ML/R/sysml.bridge.R]: spelling: `refience`

File: R4ML/R/sysml.bridge.R [hunk 6]
```diff
@@ -285,8 +285,8 @@ sysml.MLContext <- setRefClass("sysml.MLContext",
 #'
 #' @family MLContext functions
 #'
-#' @field env An R environment that stores bookkeeping states of the class
-#'        along with java ref corresponding to jvm
+#' @field env An R environment that stores bookkeeping states of the class,
+#'        along with java refierence corresponding to the JVM.
```
Comment (bdwyer2) [inline R4ML/R/sysml.bridge.R]: Spelling: `refierence`

File: R4ML/R/util.io.R [hunk 7]
```diff
@@ -212,11 +212,11 @@ is.r4ml.fs.valid <- function() {
 }
 
 #' r4ml.read.csv
-#' @description Returns a R data.frame in local mode or a SparkR DataFrame in cluster mode.
-#' @param file path to the input file
-#' @param header logical
-#' @param stringsAsFactors logical
-#' @param inferSchema logical
+#' @description Returns an R data.frame in local mode or a SparkR DataFrame in cluster mode.
```
Comment (bdwyer2) [inline R4ML/R/util.io.R]: Should we change `R data.frame` to just `data.frame` to be consistent with the rest of the the documentation?

Comment (bdwyer2) [inline R4ML/R/util.io.R]: I don't think we should say `SparkR DataFrame`. See my above comment about using either `Spark DataFrame` or `SparkDataFrame`.

File: R4ML/R/util.io.R [hunk 8]
```diff
@@ -282,16 +282,16 @@ r4ml.read.csv <- function(
...
+#' "DEBUG", "INFO", "WARN", "ERROR", "FATAL", and "OFF".
+#' Like other log services, setting a log level to WARN enables all the
 #' higher level logging (i.e ERROR FATAL) but disables lower level logging
 #' (i.e TRACE, DEBUG, INFO).
 #' NOTE: ALL and OFF are used to enable all the logs or disable all the logs
-#' respectively. Since SystemML and Spark related JVM logging level is for 
-#' higer level debug, user have different ways to control jvm and R logging 
-#' seperately.
+#' respectively. Since SystemML and Spark related JVM logging levels are for 
+#' higher level debuggging, the user is able to set JVM and R logging levels
```
Comment (bdwyer2) [inline R4ML/R/util.io.R]: spelling: `debuggging`

**Review by bdwyer2 — CHANGES_REQUESTED** [review]

File: R4ML/R/r4ml.frame.R (hunk 4 above)
Comment (nilmeier) [reply R4ML/R/r4ml.frame.R:102]: ok, I'll update.  Didn't see these.

File: R4ML/R/ml.model.pca.R (hunk 3 above)
Comment (nilmeier) [reply R4ML/R/ml.model.pca.R]: ok...will do.

File: R4ML/R/sysml.bridge.R (hunk 5 above)
Comment (nilmeier) [reply R4ML/R/sysml.bridge.R]: sorry, will fix.

File: R4ML/R/sysml.bridge.R (hunk 6 above)
Comment (nilmeier) [reply R4ML/R/sysml.bridge.R]: ok same error as above...

File: R4ML/R/util.io.R (hunk 7 above)
Comment (nilmeier) [reply R4ML/R/util.io.R]: I have no preference here...it might have been added for clarity to compare to the other data frame types.  It is the only place we describe it this way.  I'll plan to delete unless I hear otherwise.

Comment (nilmeier) [reply R4ML/R/util.io.R]: I'll go with SparkDataFrame unless I hear otherwise.

File: R4ML/R/util.io.R (hunk 8 above)
Comment (nilmeier) [reply R4ML/R/util.io.R]: got it...thx!

File: R4ML/R/ml.model.svm.R (hunk 1 above)
Comment (nilmeier) [reply R4ML/R/ml.model.svm.R]: I agree that it hf and hm are very confusing since all references to hydra have been removed.  I can change it to r4ml_df if you like that better.  The other change(rm -> r4mlMat) is okay then?

File: R4ML/R/ml.model.cox.R [hunk 9]
```diff
@@ -311,7 +311,7 @@ setMethod(f = "show", signature = "r4ml.coxph", definition =
 #' @name summary.r4ml.coxph
 #' @title Cox Summary
 #' @description Summarizes cox proportional hazard analysis
-#' @param object A cox model \code{r4ml.matrix}
+#' @param object (r4ml.matrix):  A cox model
```
Comment (bdwyer2) [inline R4ML/R/ml.model.cox.R]: The type for this should be `r4ml.coxph` not `r4ml.matrix`

File: R4ML/R/ml.model.glm.R [hunk 10]
```diff
@@ -423,17 +423,17 @@ r4ml.ml.parseFamilyAndLink <- function(family) {
...
-#' @description This method allows to score/test a GLM model for a given r4ml.matrix. If the testing set is labeled,
+#' @description This method allows one to score/test a GLM model for a given r4ml.matrix. If the testing set is labeled,
 #' testing will be done and some statistics will be computed to measure the quality of the model. Otherwise, scoring will be performed
 #' and only the predictions will be computed.
-#' @param object (r4ml.glm) a GLM model built by Big R
-#' @param data (r4ml.matrix) the testing set
-#' @param family (family) the distribution family and link function. Supported families are "gaussian", "poisson", "gamma", and "inverse.gaussian".
+#' @param object (r4ml.glm):  A GLM model built by Big R
+#' @param data (r4ml.matrix):  The test dataset
+#' @param family (family):  The distribution family and link function. Supported families are "gaussian", "poisson", "gamma", and "inverse.gaussian".
```
Comment (bdwyer2) [inline R4ML/R/ml.model.glm.R]: Maybe we should change `(family)` to `(character)`

File: R4ML/R/ml.model.svm.R (hunk 1 above)
Comment (bdwyer2) [reply R4ML/R/ml.model.svm.R]: I'm not sure what the best answer is here. I think ideally the name should all be lowercase. I've been using `r4_...` but part of me doesn't like having a number in the variable name.

File: R4ML/R/ml.model.pca.R (hunk 3 above)
Comment (nilmeier) [reply R4ML/R/ml.model.pca.R]: I remember why I moved this...we kind of converged on listing the data type after the variable.  I moved (optional) so that it doesn't look like there is a data type called "optional".

File: R4ML/R/ml.model.svm.R (hunk 1 above)
Comment (nilmeier) [reply R4ML/R/ml.model.svm.R]: yes, point taken on the number...but I'm not sure we can change the usage of r4ml throughout the code (in class names, for example).  I'll keep it as r4ml_df for now...

File: R4ML/R/ml.model.glm.R (hunk 10 above)
Comment (nilmeier) [reply R4ML/R/ml.model.glm.R]: also changed all types from (string) to (character)

File: R4ML/R/ml.model.pca.R (hunk 3 above)
Comment (nilmeier) [reply R4ML/R/ml.model.pca.R]: ok will add after type definition.

**Review by bdwyer2 — APPROVED** [review]

File: R4ML/R/ml.model.cox.R (hunk 9 above)
Comment (aloknsingh) [reply R4ML/R/ml.model.cox.R]: sorry for late reply. but isn't r4ml.coxph is a matrix? I mean from the data type perspective?

**Comment by bdwyer2** [comment]
@aloknsingh I disagree. `family` is not an R class.

**Comment by aloknsingh** [comment]
@bdwyer2 ok I see the point everything in the () is class. I think we are also documenting the family types in the next times. so in that case pl @nilmeier keep the character.

**Review by bdwyer2 — APPROVED** [review]
